tests/pytests/test_empty.py

```diff
@@ -3,28 +3,42 @@
 
 EMPTY_RESULT = [0]
 
-def testEmptyValidations():
-    """Validates the edge-cases of the `ISEMPTY` field option"""
+def testEmptyValidations(env):
+    """Validates the edge-cases of the `INDEXEMPTY` field option"""
 
-    env = Env(moduleArgs="DEFAULT_DIALECT 2")
+    MAX_DIALECT = set_max_dialect(env)
 
-    # Create an index with a TAG and a TEXT field, both of which don't index
-    # empty values
-    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'tag', 'TAG', 'text', 'TEXT').ok()
+    for dialect in range(2, MAX_DIALECT + 1):
+        env = Env(moduleArgs="DEFAULT_DIALECT " + str(dialect))
+        conn = getConnectionByEnv(env)
 
-    # Test that we get an error in case of a user tries to use "isempty(@field)"
-    # when `field` does not index empty values.
-    env.expect('FT.SEARCH', 'idx', 'isempty(@tag)').error().contains(
-        'Field `tag` should enable `ISEMPTY` in the index SCHEMA in order to support empty values'
+        # Create an index with a TAG and a TEXT field, both of which don't index
+        # empty values
+        env.expect('FT.CREATE', 'idx', 'SCHEMA', 'tag', 'TAG', 'text', 'TEXT').ok()
+
+        # Test that we get an error in case of a user tries to search for an
+        # empty value when `field` does not index empty values.
+        # TODO: For dialect < 5, we should also throw the error
```

Comment:
Can't we also throw an error for all dialects since it is not breaking - if it was causing a syntax error in older dialects, now it will explain that the `INDEXEMPTY` is needed